File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.cpp
class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        queue<int> q(students.begin(), students.end());
        for(int i = 0; i < sandwiches.size(); i++)
        {
            int count = 0;
            while(q.front() != sandwiches[i])
            {
                count++;
                if(count == q.size())
                {
                    return q.size();
                }
                q.push(q.front());
                q.pop();
            }
            q.pop();
        }
        return 0;
    }
};

File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        c = 0
        sq = 0
        for stud in students:
            if stud == 0:
                c += 1
            else:
                sq += 1
        for sand in sandwiches:
            if sand == 0:
                if c == 0:
                    break
                c -= 1
            else:
                if sq == 0:
                    break
                sq -= 1
        return c + sq
